test/src/test/java/org/corfudb/integration/ClusterReconfigIT.java

```diff
@@ -879,4 +880,67 @@ public void healNodesWorkflowTest() throws Exception {
         shutdownCorfuServer(corfuServer_2);
         shutdownCorfuServer(corfuServer_3);
     }
+
+    /**
+     * Test that a disconnected CorfuRuntime using a systemDownHandler reconnects itself once the
+     * Server is back online.
+     */
+    @Test
+    public void reconnectDisconnectedClient() throws Exception {
+        // Set up cluster of 1 node.
+        final int PORT_0 = 9000;
+        Process corfuServer_1 = runSinglePersistentServer(corfuSingleNodeHost, PORT_0);
+
+        final int systemDownHandlerLimit = 10;
+        CorfuRuntime runtime = CorfuRuntime.fromParameters(CorfuRuntimeParameters.builder()
+                .layoutServer(NodeLocator.parseString(DEFAULT_ENDPOINT))
+                .systemDownHandlerTriggerLimit(systemDownHandlerLimit)
+                .build()).connect();
+        // Register the system down handler to throw a RuntimeException.
+        runtime.registerSystemDownHandler(() -> {
+            throw new RuntimeException("SystemDownHandler");
+        });
+
+        UUID streamId = CorfuRuntime.getStreamID("testStream");
+        IStreamView stream = runtime.getStreamsView().get(streamId);
+
+        final byte[] testPayload = "test".getBytes();
+        final int appendNum = 3;
+        // Preliminary writes.
+        for (int i = 0; i < appendNum; i++) {
+            stream.append(testPayload);
+        }
+
+        // Shutdown the server.
+        shutdownCorfuServer(corfuServer_1);
+
+        final Semaphore latch = new Semaphore(1);
+        latch.acquire();
+
+        // Invalidate the layout so that there are no cached layouts.
+        runtime.invalidateLayout();
+
+        // Spawn a thread which tries to append to the stream. This should initially get stuck and
+        // trigger the systemDownHandler.
+        // This would start the server and finally wait for the append to complete.
+        Thread t = new Thread(() -> {
+            while (true) {
+                try {
+                    stream.append(testPayload);
+                    break;
+                } catch (RuntimeException sue) {
+                    // Release latch and try again..
+                    latch.release();
+                }
+            }
+        });
+        t.start();
+
+        assertThat(latch.tryAcquire(PARAMETERS.TIMEOUT_LONG.toMillis(), TimeUnit.MILLISECONDS))
+                .isTrue();
+        corfuServer_1 = runSinglePersistentServer(corfuSingleNodeHost, PORT_0);
+
+        t.join(PARAMETERS.TIMEOUT_LONG.toMillis());
```

Comment:
Should you test that the client actually reconnected and that the append completed successfully? otherwise it might have timed out as well.

Reply:
@annym The append will pass only if the append was successful. Else we will not break out from the while loop.
However, I agree we need this check. I ran it locally and asserted that the append was successful. Since I have already merged this change, I will include this assertion in the PR for master rather than 0.2.1 if that's okay.

Reply:
@zalokhan yeah that's fine with me.